packages/calypso-e2e/src/lib/blocks/block-flows/all-form-fields.ts

```diff
@@ -52,32 +59,43 @@ export class AllFormFieldsFlow implements BlockFlow {
 			[ 'Date Picker', 'Add label…' ],
 			[ 'Phone Number Field', 'Add label…' ],
 			[ 'Multi-line Text Field', 'Add label…' ],
-			[ 'Checkbox', 'Add label…' ],
+			[ 'Checkbox', isRefactor ? 'Add option…' : 'Add label…' ],
 			[ 'Multiple Choice (Checkbox)', 'Add label' ],
 			[ 'Single Choice (Radio)', 'Add label' ],
 			[ 'Dropdown Field', 'Add label' ],
 			[ 'Terms Consent', 'Add implicit consent message…' ],
 		];
 		for ( const [ blockName, accessibleLabelName ] of remainingBlocksToAdd ) {
-			await this.addFieldBlockToForm( context, blockName );
+			await this.addFieldBlockToForm( context, blockName, lastBlockName, isRefactor );
 			await labelFormFieldBlock( context.addedBlockLocator, {
 				blockName,
 				accessibleLabelName,
 				labelText: this.addLabelPrefix( blockName ),
+				isRefactor,
 			} );
+			lastBlockName = blockName;
 		}
 
 		// And we just wrap up labeling the auto-added blocks.
-		const otherBlocksToLabel = [
-			[ 'Button', 'Add text…' ],
-			[ 'Single Choice Option', 'Add option…' ],
-			[ 'Multiple Choice Option', 'Add option…' ],
-		];
-		for ( const [ blockName, accessibleLabelName ] of otherBlocksToLabel ) {
+		const otherBlocksToLabel = isRefactor
+			? [
+					[ 'Button', 'Add text…' ],
+					[ 'Option', 'Add option…', 'Single Choice (Radio)' ],
+					[ 'Option', 'Add option…', 'Multiple Choice (Checkbox)' ],
+			  ]
+			: [
+					[ 'Button', 'Add text…' ],
+					[ 'Single Choice Option', 'Add option…' ],
+					[ 'Multiple Choice Option', 'Add option…' ],
+			  ];
```

Comment:
In the refactored form fields, we re-use the same 'Label' and 'Input' and 'Option' blocks within multiple fields. To ensure we can find and label those common blocks correctly, we need to pass in the parent block (ie, we add one label to Options within Radio blocks, and another to Options with Checklist blocks).

You can see the logic for handling these added to the labelFormFieldBlock() method below.